docs/io-cloud-storage-sink.md

```diff
@@ -143,14 +143,59 @@ Before using the Cloud Storage sink connector, you need to create a configuratio
 
 ### Data format types
 
-Cloud Storage Sink Connector provides multiple options of output format, which are JSON, Avro, Bytes, or Parquet. The default format is JSON.
+Cloud Storage Sink Connector provides multiple output format options, including JSON, Avro, Bytes, or Parquet. The default format is JSON.
 With current implementation, there are some limitations for different formats:
 
-- JSON: only support structured Pulsar schema types, including `JSON`, `PROTOBUF`, `AVRO`, and `PROTOBUF_NATIVE`, will try to convert the data to JSON from `String` and `Bytes` message.
-- Avro / Parquet: only support structured Pulsar schema types, including `JSON`, `PROTOBUF`, `AVRO`, and `PROTOBUF_NATIVE`.
-- Bytes: support all Pulsar schema types.
-
-When the connector receiving a message with a schema type that is not supported by the connector, the connector will `fail` the message by default, if you want to skip the non-supported messages, you can set `skipFailedMessages` to `true`.
+| Pulsar Schema  | writer: Avro | writer: JSON | writer: Parquet | writer: Bytes |
+|----------------|--------------|--------------|-----------------|---------------|
+| Primitive      | ✗            | ✔ *          | ✗               | ✔             |
+| Avro           | ✔            | ✔            | ✔               | ✔             |
+| Json           | ✔            | ✔            | ✔               | ✔             |
+| Protobuf **    | ✔            | ✔            | ✔               | ✔             |
+| ProtobufNative | ✔ ***        | ✗            | ✔               | ✔             |
+> *: The JSON writer will try to convert the data with a `String` or `Bytes` schema to JSON-format data if convertable.
+>
+> **: The Protobuf schema is based on Avro schema, it will use Avro as an intermediate format, so it may not provide the best effort conversion.
```

Comment:
```suggestion
> **: The Protobuf schema is based on the Avro schema. It uses Avro as an intermediate format, so it may not provide the best effort conversion.
```